astropy/units/equivalencies.py

```diff
@@ -811,3 +811,31 @@ def with_H0(H0=None):
     h100_val_unit = Unit(100/(H0.to_value((si.km/si.s)/astrophys.Mpc)) * astrophys.littleh)
 
     return Equivalency([(h100_val_unit, None)], "with_H0", kwargs={"H0": H0})
+
+
+def cosmological(z, cosmology=None, comoving=False):
+    """
+    Returns the equivalence between angles and cosmological distances.
+
+    Parameters
+    ----------
+    z : float
+        the redshift to consider for the conversion.
+    cosmology : `~astropy.cosmology.FLRW`
+        The cosmology to be used. If `None`, the astropy default cosmology.
+    comoving : bool
```

Comment:
```suggestion
    comoving : bool, optional
```

Reply:
Though it might be better to implement this as

```suggestion
    distance : {'comoving', 'proper'}
```